### DISCUSSION TITLE
Pagination caching ideas

### DISCUSSION THREAD

**Discussion by lucius100** [pocketbase/pocketbase#5666] [Q&A]
so I have function that use api call from pocketbase
${POCKETBASE_URL}/api/collections/videos/records?page=${pageFromSearchParams}&perPage=12&sort=-created

I have problem with duplication in pagination which similar to scenario mentioned here

https://www.freecodecamp.org/news/how-to-implement-cacheable-pagination-of-frequently-changing-content-c8ddc8269e81/

page 1 api call, cached
page 2 api call, cached
new item added in the record
page 3 make new api call, got duplicate, cached

the reason it duplicate because new record added and new items pushed which make items in page 2 pushed into page 3, and duplicate. 

I know this is not really pocketbase related, but I want to get solution from more experienced programmer regarding pagination caching, thanks

**Comment by KelvinConnor** [comment]
You might want mention a bit more about the tech stack and architecture to get a better recommendation. I would look at Caching Repository patterns. Fortunately is it common use case many have faced, maybe there is a good solution for your language. 

For you specific situation I don't fully understand the use case where you 'know it should go into cache AND it will be potentially be retrieved in a subsequent retrieval.' It seems perhaps their is a problem with the logic. I will say in my own PB caching objects, I only update or delete items in cache. For an insert I simply clear and restart. Not sure if that helps at all.

**Reply by lucius100** [reply]
I think I will skip using cache on pagination and just cache my other pages.
